autoload/utils/exec/job.vim

```diff
@@ -51,36 +51,35 @@ function! s:createQuickFix() abort
     if l:bufnr == -1
         return
     endif
-    cexpr ''
     let l:old_error = &errorformat
     if s:err_fmt !=# ''
         let &errorformat = s:err_fmt
     endif
     execute 'cgetbuffer ' . l:bufnr
     call setqflist( [], 'a', { 'title' : s:cmake4vim_job[ 'cmd' ] } )
-    call s:closeBuffer()
     if s:err_fmt !=# ''
         let &errorformat = l:old_error
     endif
     " Remove cmake4vim job
     let s:cmake4vim_job = {}
 endfunction
 
-function! s:vimOut(channel, message) abort
-    if empty(s:cmake4vim_job) || a:channel != s:cmake4vim_job['channel']
-        return
-    endif
-    call s:appendLine(a:message)
-endfunction
-
-function! s:vimExit(channel, message) abort
-    if empty(s:cmake4vim_job) || a:channel != s:cmake4vim_job['job']
-        return
-    endif
+function! s:vimClose(channel) abort
     call s:createQuickFix()
-    if a:message != 0
-        copen
-    endif
+
+    let l:cmake4vim_window = win_findbuf(bufnr(s:cmake4vim_buf))
+
+    call win_gotoid( l:cmake4vim_window[ 0 ] )
+    copen
+    cbottom
+    mark Q
+    wincmd p
+    normal `Q
+    cclose
+    wincmd p
```

Comment:
@ilyachur this is the only remaining part. Opening `qflist` in the current buffer, and I haven't found a good way how to do that.

I want to do that because I don't always want the `qflist` to open at bottom, if I move the buffer while it's running the job, I want the `qflist` to open there, not at the bottom.

Reply:
I didn’t have time to reply to the comment, but it is already outdated :)